### PR TITLE
shipit_static_analysis: Add test to verify that our clang-tidy checks still exist

### PR DESCRIPTION
**Pull request by jankeromnes**

### CODE REVIEW

**Comment by marco-c** [comment]
When we will have integration tests, this should be part of that, so we don't only run it on commits, but also periodically.

**Comment by marco-c** [comment]
> When we will have integration tests, this should be part of that, so we don't only run it on commits, but also periodically.

Actually, probably unneeded, given that the configuration is in the project itself and the clang version too.

**Comment by sylvestre** [comment]
Please make sure that we have all the necessary info when we regress!

**Comment by jankeromnes** [comment]
It works! Artificially prefixing checkers with `oops-` shows that the error in the logs is pretty explicit:

clang-tidy-error

**Comment by jankeromnes** [comment]
@La0 or @marco-c please have a look when you can. 😄

**Comment by marco-c** [comment]
A suggestion: since we don't have integration tests yet, it would be good to make this both a test and a runtime check in the task. This way, we get notified when this happens in production even if we are not actively changing anything in the project's code.

File: src/shipit_static_analysis/tests/test_clang.py [hunk 1]
```diff
@@ -173,3 +174,38 @@ def test_clang_tidy(tmpdir, mock_config, mock_stats, mock_revision):
...
+    available_checkers = set(line.strip() for line in clang_output.split('\n'))
+
+    # Import our own clang-tidy configuration
+    from shipit_static_analysis.config import CONFIG_URL, settings
+    assert len(settings.clang_checkers) > 0, \
+        'Firefox clang-tidy configuration {} should specify > 0 clang_checkers'.format(CONFIG_URL)
+
+    # Verify that our specified clang-tidy checks actually exist
+    for checker in settings.clang_checkers:
+        name = checker['name']
```
Comment (marco-c) [inline src/shipit_static_analysis/tests/test_clang.py]: Nit: you can use [fnmatch](https://docs.python.org/3/library/fnmatch.html) to reduce the amount of code here.

**Comment by La0** [comment]
@jankeromnes If you use `fnmatch` it would be easy to add the `logger.error` in `ClangTidy` as the settings file is already downloaded during normal runtime

**Comment by jankeromnes** [comment]
Ok, I've added a runtime check (in `ClangTidy` constructor, with a new `list_available_checks` static method for common code) and the build is green again.

@La0 and/or @marco-c, please have another look when you can. 🙂

**Comment by jankeromnes** [comment]
Perfect, thanks a lot! 👍

**Pull request merged**